### ISSUE TITLE
Consider updating 'BenchmarksGame' benchmarks to use HWIntrinsics

### ISSUE THREAD

**Issue by tannergooding** [dotnet/corefx#42101]
The "BenchmarkGames" for C# have been updated to use .NET Core 3.0: https://benchmarksgame-team.pages.debian.net/benchmarksgame/fastest/csharp.html

Given that .NET Core 3.0 now supports [Hardware Intrinsics](https://devblogs.microsoft.com/dotnet/hardware-intrinsics-in-net-core/), it may be beneficial to updates these benchmarks to use them and see how much faster we can be.

We track our implementations of these toy benchmarks in the dotnet/performance repo: https://github.com/dotnet/performance/tree/master/src/benchmarks/micro/coreclr/BenchmarksGame and they would ideally also be contributed back to the Benchmark Games repo.

**Comment by tannergooding** [comment]
The contribution guide for the benchmark games repo is here: https://salsa.debian.org/benchmarksgame-team/benchmarksgame/blob/master/CONTRIBUTING.md

The information for how the programs are measured is here: https://benchmarksgame-team.pages.debian.net/benchmarksgame/how-programs-are-measured.html

**Comment by tannergooding** [comment]
CC. @danmosemsft, @adamsitnik

**Comment by KoziLord** [comment]
The benchmarks run on a Q6600 meaning that there's a few limitations as to what we can do:

We can only use up to SSE3, anything above it isn't supported on the CPU these benchmarks run on.
Apparently unaligned reads are very slow on a Q6600 so it's worth keeping it in mind.

There's definitely more things that are worth to keep in mind when creating a benchmark best suited for this.

**Comment by john-h-k** [comment]
so, for some reason that I don't think anyone really understands, these benchmarks are run on a Core 2 Q6600, from, uh, 2007. Lovely. So there's only up to SSSE3 available.

**Comment by tannergooding** [comment]
Ideally we would write a benchmark that runs well on both modern machines with AVX support and older machines with only SSE3 support 😄

**Comment by benaadams** [comment]
Also R2R run https://salsa.debian.org/benchmarksgame-team/benchmarksgame/issues/212 ?

**Comment by KoziLord** [comment]
I agree, the benchmarks should support a variety of hardware, however this definitely needs to be taken into account when writing the benchmarks (as that's the CPU that's gonna showcase Core's speed).

**Comment by danmosemsft** [comment]
I wonder whether the maintainer is looking for donations for a new machine 😄

**Comment by danmosemsft** [comment]
cc @AnthonyLloyd who has done a lot of work on the C# and F# benchmarks in the past and may have suggestions for opportunities.
